### PR TITLE
Add support for storing topic messages

### PR DESCRIPTION
**Pull request by schancel**
Adding a topic store for reading and writing broadcast messages to the rocks database.

### CODE REVIEW

File: cashweb-registry/src/store/db.rs [hunk 1]
```diff
@@ -7,11 +7,14 @@ use rocksdb::ColumnFamilyDescriptor;
 use thiserror::Error;
 
 use crate::store::metadata::DbMetadata;
+use crate::store::topics::DbTopics;
 
 // We collect the column family constants here so we have a nice overview.
 // This makes it easier to keep cf names consistent and non-conflicting.
 pub(crate) const CF_METADATA: &str = "metadata";
 pub(crate) const CF_PKH_BY_TIME: &str = "pkh_by_time";
+pub(crate) const CF_MESSAGES: &str = "messages";
```
Comment (schancel) [inline cashweb-registry/src/store/db.rs]: Might want to change these.

File: cashweb-registry/src/store/topics.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        let cf_payloads = db.cf(CF_PAYLOADS).unwrap();
+        DbTopics {
+            db,
+            cf_messages,
+            cf_payloads,
+        }
+    }
+
+    /// Put a serialized `Message` to database.
+    pub fn put_message(&self, timestamp: u64, message: &TopicPayload) -> Result<(), DbTopicsError> {
```
Comment (EyeOfPython) [inline cashweb-registry/src/store/topics.rs]: Use `bitcoinsuite_error::Result` with ErrorMeta here, then you don't have to have such a crazy error type.

File: cashweb-registry/src/store/topics.rs [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+    /// Replace a serialized `Message` to database. No need to update
+    /// indexes as they are all pointing to this entry.
+    pub fn update_message(&self, message: &TopicPayload) -> Result<(), DbTopicsError> {
+        let buf = message.to_proto().encode_to_vec();
+
+        let payload_hash = message.payload_hash().as_slice().to_vec();
+
+        self.db
+            .rocksdb()
```
Comment (EyeOfPython) [inline cashweb-registry/src/store/topics.rs:106]: You should use the `put` on `Db` directly

File: cashweb-registry/src/store/topics.rs [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    /// Get a vector of messages starting at some unix timestamp.
+    /// TODO: actually use this
+    #[allow(dead_code)]
+    pub fn get_messages(&self, topic: &str, from: i64) -> Result<Vec<TopicPayload>, DbTopicsError> {
+        self.get_messages_to(topic, from, i64::MAX)
+    }
+
+    /// Get a specific message by payload hash.
+    pub fn get_message(&self, payload_digest: &[u8]) -> Result<TopicPayload, DbTopicsError> {
+        match self.db.rocksdb().get_cf(self.cf_payloads, payload_digest)? {
```
Comment (EyeOfPython) [inline cashweb-registry/src/store/topics.rs]: Use `ok_or_else` instead of `match` here

**Review by EyeOfPython — CHANGES_REQUESTED** [review]

File: cashweb-registry/src/store/topics.rs (hunk 2 above)
Comment (schancel) [reply cashweb-registry/src/store/topics.rs]: ErrorMeta?

Comment (schancel) [reply cashweb-registry/src/store/topics.rs]: Can you elaborate on what would need to happen. You helped me setup the error type above last time.

**Pull request merged**